src/pyodide/internal/introspection.py

```diff
@@ -86,3 +87,14 @@ async def wrapper_func(relaxed, inst, prop, *args, **kwargs):
         return python_to_rpc(await result)
     else:
         return python_to_rpc(result)
+
+
+class CpuLimitExceeded(BaseException):
+    pass
+
+
+def raise_cpu_limit_exceeded(signum, frame):
+    raise CpuLimitExceeded("Python Worker exceeded CPU time limit")
+
+
+signal.signal(signal.SIGINT, raise_cpu_limit_exceeded)
```

Comment:
Bikeshed: SIGXCPU ("exceeded CPU")  could be another option. I have no strong opinion either way, though.